src/transformStateWithClones.js

```diff
@@ -1,13 +1,32 @@
 'use strict';
 
 /**
- * @param {Object} state
- * @param {Object[]} actions
+ * @param {objetoect} state
+ * @param {objetoect[]} actions
  *
- * @return {Object[]}
+ * @return {objetoect[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let objeto = { ...state };
+  const novo = [];
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+        objeto = { ...objeto, ...extraData };
+    } else if (type === 'removeProperties') {
+      for (const key of keysToRemove) {
+        delete objeto[key];
+      }
+    } else {
+      objeto = {};
+    }
```

Comment:
Instead of using an `if-else` chain, consider using a `switch` statement to handle the different types of actions. This will make the code more readable and maintainable, especially if more action types are added in the future. Also, make sure to include a `default` case for error handling as per the checklist.